src/api/routes/pox2.ts

```diff
@@ -0,0 +1,84 @@
+import * as express from 'express';
+import { asyncHandler } from '../async-handler';
+import {
+  BurnchainReward,
+  BurnchainRewardListResponse,
+  BurnchainRewardSlotHolder,
+  BurnchainRewardSlotHolderListResponse,
+  BurnchainRewardsTotal,
+} from '@stacks/stacks-blockchain-api-types';
+
+import { isValidBitcoinAddress, tryConvertC32ToBtc } from '../../helpers';
+import { InvalidRequestError, InvalidRequestErrorType } from '../../errors';
+import { parseLimitQuery, parsePagingQueryInput } from '../pagination';
+import { PgStore } from '../../datastore/pg-store';
+import { parsePox2Event } from '../controllers/db-controller';
+import { validatePrincipal, validateRequestHexInput } from '../query-helpers';
+
+const MAX_EVENTS_PER_REQUEST = 250;
+
+const parseQueryLimit = parseLimitQuery({
+  maxItems: MAX_EVENTS_PER_REQUEST,
+  errorMsg: '`limit` must be equal to or less than ' + MAX_EVENTS_PER_REQUEST,
+});
+
+export function createPox2EventsRouter(db: PgStore): express.Router {
+  const router = express.Router();
+
+  router.get(
+    '/',
+    asyncHandler(async (req, res) => {
+      const limit = parseQueryLimit(req.query.limit ?? 96);
+      const offset = parsePagingQueryInput(req.query.offset ?? 0);
+
+      const queryResults = await db.getPox2Events({ offset, limit });
+      const parsedResult = queryResults.map(r => parsePox2Event(r));
+      const response = {
+        limit,
+        offset,
+        results: parsedResult,
+      };
+      res.json(response);
+    })
+  );
+
+  // TODO: this should probably be a tx route e.g. /extended/v1/tx/:tx_id/pox2_events
```

Comment:
I agree, could we make this change? Maybe in a later PR if it's too much work

Reply:
Yeah, it's not super clear yet what the interface for exposing these should look like just yet. I think it makes sense to figure that out in a later PR.